### PR TITLE
Fix for "race condition when opening new empty buffer"

### PR DESCRIPTION
**Pull request by StevenSavold**
This is the PR for #20 

Let me know if you would rather have a different solution to the problem. Extending the critical sections like this is probably not ideal. 🤷‍♂️

### CODE REVIEW

File: src/buffer.jai [hunk 1]
```diff
@@ -503,6 +503,7 @@ find_or_create_buffer :: (path: string) -> buffer_id: s64, created: bool {
         if path.count > 248 return -1, false;  // Windows bad
     }
     lock(*open_buffers_lock);
+    defer unlock(*open_buffers_lock);  // We need the lock potentially through setting the file data, because creating a new buffer will look for an "empty" buffer
```
Comment (focus-editor) [inline src/buffer.jai:506]: This is a problem, it means that the workspace scanner can't add buffers in parallel any more. Yeah we need to find a different solution. I think maybe for now I will just allow to create a million new buffers if the user desires.

**Comment by focus-editor** [comment]
I've added this commit which solves the problem - https://github.com/focus-editor/focus/commit/880836e2d3c51a345db3bb966379f2fc5d024405

Thank you for your help!

**Comment by focus-editor** [comment]
Actually then I realised that the only thing that needed to be done was this https://github.com/focus-editor/focus/commit/51fa8bf5ce11987f1a5d4948da0b2988089b1d82

The flag is set only in the main thread so it's safe to check it

**Comment by StevenSavold** [comment]
Happy to have been some help 😄

**Pull request closed without merging**